divide.js

```diff
@@ -0,0 +1,39 @@
+/**
+Divides a series of numbers.
+@param {number} dividend - The number to be divided.
+@param {...number} divisors - The numbers to divide the dividend by.
+@returns {number} - The result of the division.
+*/
+function divide() {
+    const argumentsArray = removeNotExpectedTypesFromArguments('number', arguments)
+
+    return argumentsArray.slice(1).reduce((result, argument) => {
+        if(result === 0 && argument === 0){
+            console.error('0 can\'t be divided to 0');
+
+            return 0;
+        }
+
+        return result / argument;
+    }, argumentsArray[0]);
+};
+
+
+/**
+ * Removes arguments of types that are not allowed.
+ *
+ * @param {string} allowedType - The type of arguments that are allowed.
+ * @param {Array} arguments - The arguments to filter.
+ * @returns {Array} - The filtered arguments.
+ */
+function removeNotExpectedTypesFromArguments(allowedType, arguments){
```

Comment:
1- fonksiyon ismi "removeNotExpectedTypesFromArguments" değil "removeUnexpectedTypesFromArguments" olmalı

2- arguments parametresinin ismi global bir nesne olduğu için "args" kullanılmalı

Reply:
1-Çok mantıklı
2-Bu da mantıklı